tests/bug-reports/Bug1355.fst

```diff
@@ -0,0 +1,87 @@
+module Bug1355
+
+(***** Example 1 from the issue, still fails *****)
+
+assume val t: v:(int -> Type) -> (i:int -> v i -> Type0) -> Type
+
+irreducible type w #v #r (x:t v r) = True
+
+#push-options "--__temp_no_proj  Bug1355"
+noeq type u = 
+| U:
+  k: (i:int -> Type) ->
+  x: t k (fun _ _ -> True) ->
+  f: (i:int -> Tot (y:k i{w x})) ->
+  u
+#pop-options
+
+(***** Example 2 *****)
+
+module B   = LowStar.Buffer
+
+assume val struct (a:Type0) :Type0
+
+noeq type t2 =
+  | C2: x:int -> fld:B.pointer (struct (u:int{u < x})) -> t2
+
+
+(***** Example 3 *****)
+
+noeq type t3 = | T3 :
+  (u: (nat -> Type0)) ->
+  (f: ((x: nat & u x) -> Type0)) ->
+  (x: nat) -> (y: u x) -> (z: f (| x, y |)) -> t3
+
+
+#push-options "--__temp_no_proj  Bug1355"
```

Comment:
Fails because of bad typechecking of pattern implicits ... we discussed several possibilities including adding an ascription when typechecking patterns

Reply:
We also fixed this pattern implicits issue via #2473.